File: IT/Eccezioni/Esercizi/C++/esercizio23.cpp
/* file esercizio23.cpp */

#include <iostream>
#include <stdexcept>

using std::cin;
using std::cout;
using std::cerr;
using std::endl;
using std::string;
using std::logic_error;

int main()
{
	string user;

	while (user.empty()) {
		try {
			cout << "Scrivi il tuo nome: ";
			getline(cin, user);

			if (user.empty()) {
				throw logic_error("Nome non valido!");
			}

			cout << "Ciao, " << user << "!" << endl;
		}
		catch (logic_error& e) {
			cerr << e.what() << endl;
		}
	}

	cout << "Fine del programma." << endl;
}

File: IT/Eccezioni/Esercizi/Python/esercizio23.py
# file esercizio23.py

from __future__ import print_function
if hasattr(__builtins__, 'raw_input'): input = raw_input

user = ""

while not user:
	try:
		user = input("Scrivi il tuo nome: ")
		if not user:
			raise ValueError("Nome non valido!")

		print("Ciao, {}!".format(user))

	except ValueError as e:
		print(e)

print("Fine del programma.")

